File: 0554 brick-wall/solution.go
func leastBricks(wall [][]int) int {
    fr:=make(map[int]int)
    ma:=0
    for _,v:=range wall {
        pr:=0
        for i:=0;i<len(v)-1;i++ {
            pr+=v[i]
            fr[pr]++
            ma = max(ma,fr[pr])
        }
    }
    return len(wall)-ma
}

File: 0554 brick-wall/solution.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) {
        unordered_map<long long,int> freq;
        int maxEdges = 0;

        for (auto& row : wall) {
            long long prefix = 0;
            // skip the last brick (right edge)
            for (int i = 0; i < (int)row.size() - 1; i++) {
                prefix += row[i];
                freq[prefix]++;
                maxEdges = max(maxEdges, freq[prefix]);
            }
        }

        return (int)wall.size() - maxEdges;
    }
};
